src/components/Table/components/Row.tsx

```diff
@@ -112,8 +129,39 @@ function RowImpl<R extends Kinded, S>(props: RowProps<R>): ReactElement {
   let minStickyLeftOffset = 0;
   let expandColumnHidden = false;
 
+  // used to render the whole row when dragging with the handle
+  const ref = useRef<HTMLTableRowElement>(null);
+
   return (
-    <RowTag css={rowCss} {...others} data-gridrow {...getCount(row.id)}>
+    <RowTag
+      css={rowCss}
+      {...others}
+      data-gridrow
+      {...getCount(row.id)}
+      // these events are necessary to get the dragged-over row for the drop event
+      // and spacer styling
+      onDrop={(evt) => onDrop?.(row, evt)}
+      onDragEnter={(evt) => onDragEnter?.(row, evt)}
+      onDragOver={(evt) => onDragOver?.(row, evt)}
+      ref={ref}
+    >
+      {row.draggable && (
+        <div
```

Comment:
Thought: I kinda think we need this div to go inside one of the columns, otherwise we're throwing off the column alignment between the header row (2 divs/columns) & the table rows (3 divs/columns)...

Maybe fine to ship for now/for our usage?

We could potentially give users a `<DragHandle />` component that we just tell them "put this into your 1st column" ... or a `dragColumn` helper like in our `columns.tsx` in beam, there is a `dateColumn`, `actionColumn`, etc... That would be a place to put this code that doesn't throw off the column alignment...

Oh, yeah, the `selectColumn` and `collapseColumn` in particular handle being dropped into any table/without knowing the table's kinds...

Reply:
Good call, I added a drag handle column as you suggested. The only thing that makes this a little odd is that setting the draggable flag and providing a drop callback won't really be usable unless that column is also provided.

Reply:
> ...won't really be usable unless that column is also provided.

True...if you wanted, you could do something like "if (any row has draggable || drop callback set) && (!columns has a drag column) throw error("please add a drag column").

That sort of polished UX/DX isn't really common/expected, but pretty neat/appreciated when we do have it. :-)